solidity/ecdsa/package.json

```diff
@@ -62,7 +62,7 @@
     "typescript": "^4.5.4"
   },
   "dependencies": {
-    "@keep-network/random-beacon": ">2.0.0-dev <2.0.0-ropsten",
+    "@keep-network/random-beacon": ">2.0.0-dev. <2.0.0-ropsten",
```

Comment:
Hmm, I wonder why we need this `.`? Did this help you with regenerating the `yarn.lock`?
When I tried this locally having the dot did not make any difference when I run `yarn upgrade @keep-network/random-beacon` - my `yarn.lock` stayed with `random-beacon resolved to `2.0.0-dev.0` version...

Reply:
For ">2.0.0-dev <2.0.0-ropsten":
- removed `node_modules`, removed `yarn.lock`
- run `yarn` and it resolved to `2.0.0-dev.0`

For ">2.0.0-dev. <2.0.0-ropsten":
- did the same as above and it resolved to `2.0.0-dev.2`

Reply:
The build here also passed with a `.`. It didn't before without a `.`

Reply:
This is strange. We use the syntax without the dot across all our projects and it works fine.

Reply:
I know, but it didn't work without a `.` neither on my local, nor here in CI

Reply:
I often use https://semver.npmjs.com/ to see which packages fit semver range.
With range `>2.0.0-dev <2.0.0-ropsten` it only highlights `2.0.0-dev.0` as fitting the range. With `>2.0.0-dev. <2.0.0-ropsten` none seem to fit the range.
![image](https://user-images.githubusercontent.com/78352137/159951184-8eb902d8-f0fb-452b-85ad-e4328d31b591.png)
This is very strange...

Reply:
I believe what solved the problem in CI was updating the `yarn.lock`. For the first commit, the build failed because the dependency version for `@keep-network/random-beacon` was set to `2.0.0-dev.0`. CI started passing because you updated the version in `yarn.lock`, not because you added the dot.

For your local environment, I'm not sure why the version wasn't resolved correctly, maybe it was because of yarn cache or something? From what I've seen `yarn upgrade` resolves the latest version correctly for the range we define usually.

Reply:
https://classic.yarnpkg.com/en/docs/dependency-versions/#toc-x-ranges

After the `.` if you leave empty it resolves to `*`.

Reply:
I did a clean clone `git clone` of `main` branch of `keep-core`, so no cache no nothing. I changed
```
"@keep-network/random-beacon": "../random-beacon"
```

to 

```
"@keep-network/random-beacon": ">2.0.0-dev <2.0.0-ropsten"
```

and it didn't resolve to 2.0.0-dev.2, but stayed as -dev.0

when I changed to
```
"@keep-network/random-beacon": ">2.0.0-dev.* <2.0.0-ropsten"
```
it resolved to -dev.2 

I'm not married to this solution, so if you have a better idea let's explore, but I'm not sure how to fix it differently.

Reply:
> https://classic.yarnpkg.com/en/docs/dependency-versions/#toc-x-ranges
> 
> After the `.` if you leave empty it resolves to `*`.


It may be yarn specific, I'm not sure it works the same way with npm.
If we decide to use it it would be probably better to replace the range with `"@keep-network/random-beacon": "2.0.0-dev.*"` if it works correctly.

But still, I'm not sure I'd like to change the approach.

Reply:
`2.0.0-dev.0` was tagged as `latest`. I updated the `latest` tag to point to `2.0.0.-dev.2` and I believe now it resolves to this version. Could you please confirm?

Reply:
> It may be yarn specific, I'm not sure it works the same way with npm.

idk, but we use `yarn` in this project. There had to be probably a major problem with `yarn` to make a decision to go back with `npm`. Besides `npm` might also have some things that won't be compatible with `yarn` or some other dependency manager. In either case, we don't know that for sure. Sometimes it's hard to be very generic.

> If we decide to use it it would be probably better to replace the range with "@keep-network/random-beacon": "2.0.0-dev.*" if it works correctly.

https://github.com/keep-network/keep-core/pull/2878/commits/269f249995f23025e425ae6b2f07c632fc19ac26

> But still, I'm not sure I'd like to change the approach.

I'd also rather just leave as-is and move on with other things, but `>2.0.0-dev <2.0.0-ropsten` didn't work for me.

Reply:
I tested how it resolves with another package that has a lot more versions:
`yarn add "@keep-network/keep-core"@">1.8.0-pre <1.8.0-rc"`
And it resolved all fine to `1.8.0-pre.16`.

Reply:
> `2.0.0-dev.0` was tagged as `latest`. I updated the `latest` tag to point to `2.0.0.-dev.2` and I believe now it resolves to this version. Could you please confirm?

Yes, that thing worked. [23876a4](https://github.com/keep-network/keep-core/pull/2878/commits/23876a44e9f7cce01db27d3195062de5407022d5)